File: src/main/java/org/example/array/leetcode/001-Two_Sum.java
package org.example.array.leetcode;

import java.util.Arrays;
import java.util.stream.IntStream;

class Two_Sum {
    public int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            int finalI = i;
            int index = IntStream.range(0, nums.length)
                    .filter(j -> nums[j] == complement && j != finalI)
                    .findFirst()
                    .orElse(-1);
            if (index != -1) {
                return new int[]{i, index};
            }
        }
        return new int[0];
    }
}

class Main1 {
    public static void main(String[] args) {
        Two_Sum problem = new Two_Sum();
        int[] nums = {3, 3};
        int target = 6;
        System.out.println(Arrays.toString(problem.twoSum(nums, target)));
    }
}

File: src/main/java/org/example/array/leetcode/001-Two_Sum.kt
package org.example.array.leetcode

class Two_Sum_Kotlin {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        for (i in nums.indices) {
            val complement = target - nums[i]
            val finalI = i
            val index = nums.indexOfFirst { it == complement && nums.indexOf(it) != finalI }
            if (index != -1) {
                return intArrayOf(i, index)
            }
        }
        return intArrayOf(0)
    }
}

fun main() {
    val problem = Two_Sum()
    val nums = intArrayOf(3, 3)
    val target = 6
    println(problem.twoSum(nums, target).contentToString())
}
